### PR TITLE
Add reference key and dependency image completions

### PR DESCRIPTION
**Pull request by nadundesilva**
## Automation tests
 - Unit tests 
   > N/A
 - Integration tests
   > N/A

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: components/blangserver-plugins/src/main/java/io/cellery/tooling/ballerina/langserver/plugins/Constants.java [hunk 1]
```diff
@@ -18,18 +18,55 @@
...
 public class Constants {
-
     public static final String CELLERY_PACKAGE_ORG_NAME = "celleryio";
     public static final String CELLERY_PACKAGE_NAME = "cellery";
+    public static final String CELLERY_GET_REFERENCE_METHOD = "getReference";
+
+    public static final String CELLERY_IMAGE_EXTENSION = ".zip";
+    public static final String LOCAL_REPO_DIRECTORY = System.getProperty("user.home") + File.separator + ".cellery"
+            + File.separator + "repo";
+    public static final String CELLERY_IMAGE_REFERENCE_FILE = "artifacts/ref/reference.json";
```
Comment (azinneera) [inline components/blangserver-plugins/src/main/java/io/cellery/tooling/ballerina/langserver/plugins/Constants.java]: We can use` File.separator` or replace all path resolutions with `java.nio.file.Paths#get`

File: components/blangserver-plugins/src/main/java/io/cellery/tooling/ballerina/langserver/plugins/images/ImageManager.java [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+         * @return The digest of the Cellery Image
+         */
+        private byte[] getCurrentDigest() {
+            byte[] digest = null;
+            if (imageFile.exists()) {
+                try {
+                    MessageDigest md5 = MessageDigest.getInstance("MD5");
+                    digest = md5.digest(IOUtils.toByteArray(new FileInputStream(imageFile)));
+                } catch (IOException | NoSuchAlgorithmException e) {
+                    digest = null;
```
Comment (azinneera) [inline components/blangserver-plugins/src/main/java/io/cellery/tooling/ballerina/langserver/plugins/images/ImageManager.java:269]: Shouldn't we log the error here?

File: components/blangserver-plugins/src/main/java/io/cellery/tooling/ballerina/langserver/plugins/Constants.java (hunk 1 above)
Comment (nadundesilva) [reply components/blangserver-plugins/src/main/java/io/cellery/tooling/ballerina/langserver/plugins/Constants.java]: These are not user's directory paths. These are zip entries. We access the zip entry itself directly without extracting the zip file.

According to [Zip File Spec](https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT) under 4.4.17.1, all slashes must be "/" character. So we need to stick with this constant.

Comment (nadundesilva) [reply components/blangserver-plugins/src/main/java/io/cellery/tooling/ballerina/langserver/plugins/Constants.java]: I have renamed the constant to indicate that it is a zip entry to improve readability

File: components/blangserver-plugins/src/main/java/io/cellery/tooling/ballerina/langserver/plugins/images/ImageManager.java (hunk 2 above)
Comment (nadundesilva) [reply components/blangserver-plugins/src/main/java/io/cellery/tooling/ballerina/langserver/plugins/images/ImageManager.java:269]: Added a warning log.

**Review by azinneera — APPROVED** [review]

**Comment by nadundesilva** [comment]
This completes the following tasks in #5
* Add reference key completions
* Add image list completions

**Pull request merged**